charts/server/templates/servicemonitor.yaml

```diff
@@ -0,0 +1,22 @@
+{{- if and .Values.prometheus.servicemonitor.enabled ( .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" ) }}
```

Comment:
This also requires `metrics: true` AFAICS - so I guess adding it as a condition here would be good.